components/core/src/clp/streaming_archive/writer/Archive.hpp

```diff
@@ -316,6 +321,11 @@ class Archive {
     std::vector<File*> m_files_with_timestamps_in_segment;
     std::vector<File*> m_files_without_timestamps_in_segment;
 
+    // Files in this collection only hold metadata. Files are added to this collection after
```

Comment:
maybe move this comment to where the m_file_metadata is updated?
```suggestion
    // Files in this collection only hold metadata. Files are added to this collection after
```